Mathlib/Algebra/Polynomial/AlgebraMap.lean

```diff
@@ -557,11 +570,22 @@ lemma comp_X_add_C_eq_zero_iff : p.comp (X + C t) = 0 ↔ p = 0 :=
 
 lemma comp_X_add_C_ne_zero_iff : p.comp (X + C t) ≠ 0 ↔ p ≠ 0 := comp_X_add_C_eq_zero_iff.not
 
+lemma dvd_comp_iff_of_degree_one (p q r : R[X]) (hd : r.degree = 1) (hl : IsUnit r.leadingCoeff) :
+  p ∣ q.comp r ↔ p.comp (C hl.unit.inv * (X - C (r.coeff 0))) ∣ q := by
```

Comment:
I think in this lemma it might be better to replace `r` by `C a * X + C b` and use the inverse given by an `[Invertible a]` assumption.

Reply:
Thank you very much! I felt very bad when I was writing this `hl.unit.inv`. Your advice is much better.